src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -545,6 +554,71 @@ private ImmutableArray<AdditionalTextFile> ResolveAdditionalFilesFromArguments(L
             return builder.ToImmutableArray();
         }
 
+        private static void ReportAnalyzerExecutionTime(TextWriter consoleOutput, AnalyzerDriver analyzerDriver, CultureInfo culture, bool isConcurrentBuild)
+        {
+            Debug.Assert(analyzerDriver.AnalyzerExecutionTimes != null);
+            if (analyzerDriver.AnalyzerExecutionTimes.IsEmpty)
+            {
+                return;
+            }
+
+            var totalAnalyzerExecutionTime = analyzerDriver.AnalyzerExecutionTimes.Sum(kvp => kvp.Value.TotalSeconds);
+            Func<double, string> getFormattedTime = d => d.ToString("##0.000", culture);
+            consoleOutput.WriteLine();
+            consoleOutput.WriteLine(string.Format(CodeAnalysisResources.AnalyzerTotalExecutionTime, getFormattedTime(totalAnalyzerExecutionTime)));
+
+            if (isConcurrentBuild)
+            {
+                consoleOutput.WriteLine(CodeAnalysisResources.MultithreadedAnalyzerExecutionNote);
+            }
+
+            var analyzersByAssembly = analyzerDriver.AnalyzerExecutionTimes
+                .GroupBy(kvp => kvp.Key.GetType().GetTypeInfo().Assembly)
+                .OrderByDescending(kvp => kvp.Sum(entry => entry.Value.Ticks));
+
+            consoleOutput.WriteLine();
+
+            getFormattedTime = d => d < 0.001 ?
+                string.Format(culture, "{0,8:<0.000}", 0.001) :
+                string.Format(culture, "{0,8:##0.000}", d);
+            Func<int, string> getFormattedPercentage = i => string.Format("{0,5}", i < 1 ? "<1" : i.ToString());
```

Comment:
Technically you should pass the culture into ```i.ToString()```. (For the thousands separator or negative values). i.e.
```i.ToString(culture)```